### PR TITLE
adding go version and std_package_list to releaser

### PR DESCRIPTION
**Pull request by linzhp**
**What type of PR is this?**
Feature

**What package or component does this PR mostly affect?**

tools/releaser

**What does this PR do? Why is it needed?**
According to the [release process](https://github.com/bazelbuild/bazel-gazelle/wiki/Release-process)  we need to update `language/go/std_package_list.go` and use the latest Go version to run `go mod tidy`. This PR adds the following to the releaser:
* update `language/go/std_package_list.go` as part of the release
* make releaser takes a mandatory Go version so we can update go.mod file accordingly

### CODE REVIEW

File: tools/releaser/main.go [hunk 1]
```diff
@@ -43,13 +46,15 @@ func main() {
...
 func run(ctx context.Context, stderr *os.File) error {
 	var (
-		verbose bool
+		verbose   bool
+		goVersion string
 	)
 
 	flag.BoolVar(&verbose, "verbose", false, "increase verbosity")
 	flag.BoolVar(&verbose, "v", false, "increase verbosity (shorthand)")
+	flag.StringVar(&goVersion, "go", "", "go version for go.mod")
```
Comment (xytan0056) [inline tools/releaser/main.go:55]:
```suggestion
	flag.StringVar(&goVersion, "go_version", "", "go version for go.mod")
```

File: tools/releaser/main.go [hunk 2]
```diff
@@ -182,7 +198,11 @@ This utility is intended to handle many of the steps to release a new version.
 		return err
 	}
 
-	cmd = exec.CommandContext(ctx, "bazel", "build", "//language/proto:known_imports", "//language/proto:known_proto_imports", "//language/proto:known_go_imports")
+	cmd = exec.CommandContext(ctx, "bazel", "build",
+		"//language/proto:known_imports",
+		"//language/proto:known_proto_imports",
+		"//language/proto:known_go_imports",
+		"//language/go:std_package_list")
```
Comment (xytan0056) [inline tools/releaser/main.go:205]:
```suggestion
		"//language/go:std_package_list",
	)
```

File: tools/releaser/main.go [hunk 3]
```diff
@@ -220,11 +244,17 @@ This utility is intended to handle many of the steps to release a new version.
 	return nil
 }
 
-func copyHelper(destPath, srcPath string) error {
+func copyFile(destPath, srcPath string) error {
```
Comment (xytan0056) [inline tools/releaser/main.go:247]:
```suggestion
func copyFile(workspace, destPath, srcPath string) error {
...
   path.Join(Dir...)
```

**Review by xytan0056 — COMMENTED** [review]
commend

**Review by xytan0056 — APPROVED** [review]

File: tools/releaser/main.go (hunk 3 above)
Comment (dnathe4th) [reply tools/releaser/main.go]: I'd go the other way. Change the signature to `copyFile(path string) error` and then construct `src := path.Join(os.Getenv("BUILD_WORKSPACE_DIRECTORY"), "bazel-bin", path)` and `dest := path.Join(os.Getenv("BUILD_WORKSPACE_DIRECTORY"), path)`. No reason to overly generalize a bespoke helper.

**Review by dnathe4th — APPROVED** [review]

**Review by dnathe4th — APPROVED** [review]

**Pull request merged**